PWGHF/TableProducer/treeCreatorDplusToPiKPi.cxx

```diff
@@ -0,0 +1,430 @@
+                   aod::BigTracksPID const&)
+  {
+    // Filling event properties
+    rowCandidateFullEvents.reserve(collisions.size());
+    for (auto const& collision : collisions) {
+      fillEvent(collision, 0, 1);
+    }
+
+    // Filling candidate properties
+    rowCandidateFull.reserve(candidates.size());
+    for (auto const& candidate : candidates) {
+      auto prong0 = candidate.prong0_as<aod::BigTracksPID>();
+      auto prong1 = candidate.prong1_as<aod::BigTracksPID>();
+      auto prong2 = candidate.prong2_as<aod::BigTracksPID>();
+      double yD = yDplus(candidate);
+      double eD = eDplus(candidate);
+      double ctD = ctDplus(candidate);
+      fillTable(candidate, prong0, prong1, prong2, candidate.isSelDplusToPiKPi(), invMassDplusToPiKPi(candidate), ctD, yD, eD, 0, 0);
+    }
+  }
+
+  PROCESS_SWITCH(HfTreeCreatorDplusToPiKPi, processData, "Process data", true);
+
+  void processMc(aod::Collisions const& collisions,
+                 aod::McCollisions const&,
+                 soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfCand3ProngMcRec, aod::HfSelDplusToPiKPi>> const& candidates,
+                 soa::Join<aod::McParticles, aod::HfCand3ProngMcGen> const& particles,
+                 aod::BigTracksPID const&)
+  {
+    // Filling event properties
+    rowCandidateFullEvents.reserve(collisions.size());
+    for (auto const& collision : collisions) {
+      fillEvent(collision, 0, 1);
+    }
+
+    // Filling candidate properties
+    rowCandidateFull.reserve(candidates.size());
+    for (auto const& candidate : candidates) {
+      auto prong0 = candidate.prong0_as<aod::BigTracksPID>();
+      auto prong1 = candidate.prong0_as<aod::BigTracksPID>();
+      auto prong2 = candidate.prong0_as<aod::BigTracksPID>();
+      double yD = yDplus(candidate);
+      double eD = eDplus(candidate);
+      double ctD = ctDplus(candidate);
+      fillTable(candidate, prong0, prong1, prong2, candidate.isSelDplusToPiKPi(), invMassDplusToPiKPi(candidate), ctD, yD, eD, candidate.flagMcMatchRec(), candidate.originMcRec());
+    }
+
+    // Filling particle properties
+    rowCandidateFullParticles.reserve(particles.size());
+    for (auto const& particle : particles) {
+      if (TESTBIT(std::abs(particle.flagMcMatchGen()), DecayType::DplusToPiKPi)) {
+        rowCandidateFullParticles(
+          particle.mcCollision().bcId(),
+          particle.pt(),
+          particle.eta(),
+          particle.phi(),
+          RecoDecay::y(array{particle.px(), particle.py(), particle.pz()}, RecoDecay::getMassPDG(particle.pdgCode())),
+          particle.flagMcMatchGen(),
+          particle.originMcGen());
+      }
+    }
+  }
+
+  PROCESS_SWITCH(HfTreeCreatorDplusToPiKPi, processMc, "Process MC", false);
+
+  void processForMLTraining(soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfCand3ProngMcRec, aod::HfSelDplusToPiKPi>> const& candidates,
```

Comment:
```suggestion
  void processReducedTableMC(soa::Filtered<soa::Join<aod::HfCand3Prong, aod::HfCand3ProngMcRec, aod::HfSelDplusToPiKPi>> const& candidates,
```
I would use *reduced table* instead of *train table* and *ML training* which is in general more clear IMHO. As you state in the task description, these tables can be used for other things than ML training.

Reply:
Ciao @fcatalan92 ! Thanks for the feedback.

I chose `Lite` instead of `Reduced` (as the latter is already used in the reduced workflow).

Indeed, as we need to implement it for data and for MC, I defined a configurable (which was Fabrizio's initial suggestion :D) to fill this lite table.

Reply:
Now with the configurable, you are always producing the large table even if you want only the small one. I would make this exclusive filling only one table at a time (this should have benefits in terms of memory and disk consumption).

Reply:
I fully agree, thanks :)